pkg/csaf/csaf.go

```diff
@@ -166,3 +166,52 @@ func (branch *ProductBranch) FindProductIdentifier(helperType, helperValue strin
 
 	return nil
 }
+
+type ProductList []Product
+
+// Add adds a prodocut to the product list if its not there, matching id and
+// software identifiers.
+func (pl *ProductList) Add(p Product) {
```

Comment:
Not a huge deal, but it might be better here to have this method act on a non-pointer type, and return the updated `ProductList` instead. If I'm reading this right, it would be easy for someone to have a `nil` slice (i.e. a slice's default value), in which case calling this method will panic.

Reply:
Sounds good ill update it in a follow up :) Thanks!